akka-actor/src/main/scala/akka/pattern/BackoffOptions.scala

```diff
@@ -299,6 +298,15 @@ private[akka] sealed trait ExtendedBackoffOptions[T <: ExtendedBackoffOptions[T]
    */
   def withReplyWhileStopped(replyWhileStopped: Any): T
 
+  /**
+   * Returns a new BackoffOptions with a custom handler for messages that the supervisor receives while its child is stopped.
+   * By default, a message received while the child is stopped is forwarded to `deadLetters`.
+   * Essentially, this handler replaces `deadLetters` allowing to implement custom handling instead of a static reply.
+   *
+   * @param handler PartialFunction of the received message and sender
+   */
+  def withHandlerWhileStopped(handler: ActorRef): T
```

Comment:
One thing with going for an actorRef instead of props is that it will be a shared actor, for example when used with sharding you will have to spawn a single alternative actor for an entire node. Not sure if that is a problem  or if that is ok.

Nice to not have to manage lifecycle of the stopped handler though. Hmm.

Reply:
In the end it's up to the user. You can have one actor for a node (which in most cases would probably suffice), or spawn one per supervisor as sibling if you need that.

Reply:
In the end it's up to the user. You can have one actor for a node (which in most cases would probably suffice), or spawn one per supervisor as sibling if you need that.

Reply:
Ah, you could indeed just put another actor in between. I think that is good enough.